### PR TITLE
Adding GetPipelineTestReport

### PR DESCRIPTION
**Pull request by avanloo**
see: https://docs.gitlab.com/ee/api/pipelines.html#get-a-pipelines-test-report

### CODE REVIEW

**Review by svanharmelen — APPROVED** [review]
Thanks @avanloo! 👍🏻

File: pipelines.go [hunk 1]
```diff
@@ -80,6 +80,43 @@ func (p Pipeline) String() string {
...
+	TotalCount   int                            `json:"total_count"`
+	SuccessCount int                            `json:"success_count"`
+	FailedCount  int                            `json:"failed_count"`
+	SkippedCount int                            `json:"skipped_count"`
+	ErrorCount   int                            `json:"error_count"`
+	TestSuites   []PipelintTestReportTestSuites `json:"test_suites"`
+}
+
+//PipelintTestReportTestSuites contains test suites test run in the pipeline
+type PipelintTestReportTestSuites struct {
```
Comment (svanharmelen) [inline pipelines.go:95]: Hmm... When looking at it again, maybe we can make the names just a little bit shorter:
```suggestion
type PipelintTestSuites struct {
```

File: pipelines.go [hunk 2]
```diff
@@ -80,6 +80,43 @@ func (p Pipeline) String() string {
...
+	TotalCount   int                           `json:"total_count"`
+	SuccessCount int                           `json:"success_count"`
+	FailedCount  int                           `json:"failed_count"`
+	SkippedCount int                           `json:"skipped_count"`
+	ErrorCount   int                           `json:"error_count"`
+	TestCases    []PipelineTestReportTestCases `json:"test_cases"`
+}
+
+//PipelineTestReportTestCases contains testcases test run in the pipeline
+type PipelineTestReportTestCases struct {
```
Comment (svanharmelen) [inline pipelines.go:107]: And:
```suggestion
type PipelineTestCases struct {
```

**Comment by svanharmelen** [comment]
I'll update those struct names myself, so will merge this one now. Thanks 👍🏻

**Comment by avanloo** [comment]
Let me help out with that and add also tests ;-)

**Comment by svanharmelen** [comment]
No worries, already done. Thanks!

**Comment by svanharmelen** [comment]
Can of course still add tests if you like 👍🏻

**Pull request merged**